### PR TITLE
Added fuctionality that removes item from cart when count is decrease…

### LINKED ISSUE #19: Delete automatically on Carts page
#### Description

Delete automatically on Carts page when user decreases qty to zero as shown in ss below.

Screenshot from 2022-08-19 20-57-32

#### Environment:
- Host OS: 
- Browser:

### PR DESCRIPTION
**Pull request by WattBowers**
This PR fixes #19

**Notes for Reviewers**

Added some functionality from the REMOVE reducer into the DECREASE reducer.

### CODE REVIEW

File: src/reducers/basketReducer.js [hunk 1]
```diff
@@ -98,30 +98,42 @@ export default (state = initialState, action) => {
...
                 productSelected.numbers -= 1;
-                newCartCost =
-                    state.cartCost - state.products[action.payload].price;
+                newCartCost = state.cartCost - state.products[action.payload].price;
                 newBasketNumbers = state.basketNumbers - 1;
-            }
+                //Removing item from cart in case that the ocunt in decreased to zero.
+                if(productSelected.numbers === 0) {
+                    productSelected = { ...state.products[action.payload] };
+                    let numbersBackup = productSelected.numbers;
```
Comment (kamalbuilds) [inline src/reducers/basketReducer.js:111]: Can you list why you are creating a numbersBackup

**Review by kamalbuilds — COMMENTED** [review]
Please document your changes, will be helpful in Readme

File: src/reducers/basketReducer.js (hunk 1 above)
Comment (WattBowers) [reply src/reducers/basketReducer.js:111]: Now that I am looking at it, you don't need to. That was an accidental carryover from the other method.

**Comment by kamalbuilds** [comment]
@WattBowers Please update the files

**Review by kamalbuilds — APPROVED** [review]
LGTM , we will fix the numbers backup hereafter ✔️ thanks @WattBowers

**Pull request merged**